can not install in Ubuntu 20.04.3 LTS x86_64

**Describe the bug**
[ERROR] tar: Error is not recoverable

**To Reproduce**
Steps to reproduce the behavior:
1. Install the app by running this command `bash <(curl -sL https://raw.githubusercontent.com/everledger/fabkit/master/install.sh)`
2. Where would you like to install Fabkit? [/home/zoomq/.fabkit] /home/zoomq/.fabkit
4. See error

[ERROR] tar: Error is not recoverable: exiting now
[ERROR] There was an error extracting fabkit-latest.zip into /home/zoomq/.fabkit
❌

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**

![](https://ipic.zoomquiet.top/2021-12-08-zshot%202021-12-08%2021.58.37.jpg)

**Logs**
Clean your terminal, run the same command in debug mode (`-d, --debug`) and paste the whole content of your shell (please wrap the text in a "code snippet" by using opening and closing with 3 backticks).

Copy and paste also the result of `cat ${FABKIT_ROOT}/.log`.

2021-12-08 13:55:16 UTC [ERROR] tar: Error is not recoverable: exiting now
2021-12-08 13:55:16 UTC [ERROR] There was an error extracting fabkit-latest.zip into /home/zoomq/.fabkit
2021-12-08 13:55:42 UTC [ERROR] There was an error extracting fabkit-latest.zip into /home/zoomq/.fabkit
2021-12-08 13:55:42 UTC [ERROR] tar: Error is not recoverable: exiting now
2021-12-08 14:03:06 UTC [ERROR] tar: Error is not recoverable: exiting now
2021-12-08 14:03:06 UTC [ERROR] There was an error extracting fabkit-latest.zip into /home/zoomq/.fabkit

**Config**
Paste here the result of `cat ${FABKIT_ROOT}/.lastrun`.

> not yet there...

**Environment**
Be as specific as possible!
```
OS: Ubuntu 20.04.3 LTS x86_64
Host: Droplet 20171212
Kernel: 5.4.0-91-generic
Uptime: 1 day, 10 hours, 16 mins
Packages: 1220 (dpkg)
Shell: bash 5.0.17
Terminal: /dev/pts/0
CPU: DO-Premium-Intel (2) @ 2.494GHz
GPU: 00:02.0 Red Hat, Inc. QXL paravirtual graphic card
Memory: 766MiB / 1987MiB
```

$ docker --version
Docker version 20.10.11, build dea9396

$ docker-compose --version
docker-compose version 1.25.0, build unknown

**Additional context**
Add any other context about the problem here.

--- a/install.sh
+++ b/install.sh
@@ -10,9 +10,6 @@ FABKIT_DOCKER_IMAGE="everledgerio/fabkit"
 FABKIT_LOGFILE=".fabkit.log"
 # TODO: Retrieve latest available version from web
 FABKIT_VERSION="latest"
-FABKIT_BRANCH="master"
-FABKIT_MASTER_TARBALL_URI="https://github.com/everledger/fabkit/archive/refs/heads/master.zip"
-FABKIT_TARBALL="fabkit-${FABKIT_VERSION}.zip"
 ALIASES=("fabkit" "fk")
 
 __error() {
@@ -38,6 +35,13 @@ __error() {
 __exit_on_error() {
     echo "Check the log file for more details: $(yellow "cat $FABKIT_LOGFILE")"
     kill 0
+}
+
+__check_deps() {
+    if ! type -p git &>/dev/null; then
+        __error "git required but not installed"
+        exit 1
+    fi
 }
 
 __overwrite_line() {
@@ -96,13 +100,7 @@ __setup() {
     echo
 }
 
-__download_and_extract() {
-    (
-        echo -n "Downloading $(cyan "$FABKIT_TARBALL")"
-        curl -sL $FABKIT_MASTER_TARBALL_URI -o "$FABKIT_TARBALL"
-    ) &
-    __spinner
-
+__install() {
     unset FABKIT_ROOT
     while [[ (-n "$yn" && ! "$yn" =~ ^(Y|y)) || -z "$FABKIT_ROOT" || (-n "$FABKIT_ROOT" && ! -d "$FABKIT_ROOT") ]]; do
         read -rp "Where would you like to install Fabkit? [$(yellow "$FABKIT_DEFAULT_PATH")] " FABKIT_ROOT
@@ -134,7 +132,7 @@ __download_and_extract() {
 
     if [ -w "$FABKIT_ROOT" ]; then
         (
-            if echo -n "Installing into $(cyan "${FABKIT_ROOT}")" && (__unpack_tarball 2>&1 >/dev/null | grep -iE "erro|pani|fail|fatal") > >(__error >&2); then
+            if echo -n "Downloading and installing into $(cyan "${FABKIT_ROOT}")" && (__fetch 2>&1 >/dev/null | grep -iE "erro|pani|fail|fatal") > >(__error >&2); then
                 __error "There was an error extracting ${FABKIT_TARBALL} into ${FABKIT_ROOT}"
                 exit 1
             fi
@@ -149,10 +147,8 @@ __download_and_extract() {
     fi
 }
 
-__unpack_tarball() {
-    local temp_dir=".__fabkit_tmp"
-    mkdir "$temp_dir" && tar -C "$temp_dir" -xzf "$FABKIT_TARBALL" && mv {${temp_dir}/fabkit-master/*,${temp_dir}/fabkit-master/.*} "${FABKIT_ROOT}/"
-    rm -rf "$temp_dir" "${FABKIT_TARBALL}" &>/dev/null
+__fetch() {
+    git clone https://github.com/everledger/fabkit.git ${FABKIT_ROOT}
 }
 
 __set_installation_type() {
@@ -269,19 +265,17 @@ read -n 1 -s -r -p "Press any key to start! (or CTRL-C to exit) "
 echo
 echo
 
+__check_deps
+
 if git remote -v 2>/dev/null | grep -q "fabkit"; then
     cyan "Looks like you want to run Fabkit from inside its repository path."
     echo
-    if (git checkout ${FABKIT_BRANCH} 2>&1 >/dev/null | grep -iE "erro|pani|fail|fatal") > >(__error >&2); then
-        __error "Error checking out branch ${FABKIT_BRANCH}"
-        exit 1
-    fi
     echo "Setting to $(cyan "${FABKIT_VERSION}") version."
     echo "Going to set - $(cyan "${PWD}") - as your installation path!"
     echo
     FABKIT_ROOT="$PWD"
 else
-    __download_and_extract
+    __install
 fi
 
 sleep 1

--- a/scripts/utils.sh
+++ b/scripts/utils.sh
@@ -12,13 +12,6 @@ __check_version() {
     local supported_version=($(echo -e "${2//./\\n}"))
     local installed_version=($(echo -e "${3//./\\n}"))
 
-    if ! type -p "$cmd" &>/dev/null; then
-        if (docker run --rm -i "$FABKIT_DOCKER_IMAGE" bash -c "type -p $cmd" 2>&1 >/dev/null | grep -iE "erro|pani|fail|fatal") > >(__throw >&2); then
-            logerr "${cmd} is not installed. Version >= $2 is required"
-            exit 1
-        fi
-    fi
-
     for i in "${!supported_version[@]}"; do
         if [[ "${installed_version[$i]//[aA-zZ]/}" -gt "${supported_version[$i]}" ]]; then
             break
@@ -31,10 +24,22 @@ __check_version() {
 
 __check_dep_version() {
     __check_docker_version
+    __check_docker_compose_version
     __check_bash_version
 }
 
+__check_dep() {
+    local dep=$1
+
+    if ! type -p "$dep" &>/dev/null; then
+        logerr "${dep} is not installed."
+        exit 1
+    fi
+}
+
 __check_bash_version() {
+    __check_dep bash
+
     # shellcheck disable=SC2155
     local version="$(__run "$FABKIT_ROOT" bash --version 2>/dev/null | head -n 1 | cut -d ' ' -f 4 | cut -d '(' -f 1)"
     # quick fix for non-ascii encoding
@@ -63,15 +68,8 @@ __check_java_version() {
     __check_version java "$FABKIT_JAVA_VERSION_SUPPORTED" "$version"
 }
 
-__check_docker() {
-    if ! type -p docker &>/dev/null; then
-        logerr "docker is not installed. Version >= $FABKIT_DOCKER_VERSION_SUPPORTED is required"
-        exit 1
-    fi
-}
-
 __check_docker_version() {
-    __check_docker
+    __check_dep docker
 
     if docker info --format '{{json .}}' | grep -q "Cannot connect"; then
         logerr "Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?"
@@ -81,8 +79,13 @@ __check_docker_version() {
     # shellcheck disable=SC2155
     local version=$(docker version --format '{{.Server.Version}}' 2>/dev/null)
     __check_version docker "$FABKIT_DOCKER_VERSION_SUPPORTED" "$version"
+}
+
+__check_docker_compose_version() {
+    __check_dep docker-compose
+
     # shellcheck disable=SC2155
-    version=$(docker-compose version --short 2>/dev/null)
+    local version=$(docker-compose version --short 2>/dev/null)
     __check_version docker-compose "$FABKIT_DOCKER_COMPOSE_VERSION_SUPPORTED" "$version"
 }
 

--- a/scripts/utils.sh
+++ b/scripts/utils.sh
@@ -30,13 +30,17 @@ __check_version() {
 }
 
 __check_dep_version() {
-    __check_bash_version
     __check_docker_version
+    __check_bash_version
 }
 
 __check_bash_version() {
     # shellcheck disable=SC2155
-    local version="$(__run "$FABKIT_ROOT" bash --version 2>/dev/null | cut -d ' ' -f 4 | head -n 1 | cut -d '(' -f 1)"
+    local version="$(__run "$FABKIT_ROOT" bash --version 2>/dev/null | head -n 1 | cut -d ' ' -f 4 | cut -d '(' -f 1)"
+    # quick fix for non-ascii encoding
+    if [ -z "$version" ]; then
+        version="$(__run "$FABKIT_ROOT" bash --version 2>/dev/null | head -n 1 | cut -d ' ' -f 3 | cut -d '(' -f 1)"
+    fi
     __check_version bash "$FABKIT_BASH_VERSION_SUPPORTED" "$version"
 }
 
@@ -55,11 +59,20 @@ __check_node_version() {
 
 __check_java_version() {
     # shellcheck disable=SC2155
-    local version=$(__run "$FABKIT_ROOT" java --version 2>/dev/null | cut -d ' ' -f 2 | head -n 1)
+    local version=$(__run "$FABKIT_ROOT" java --version 2>/dev/null | head -n 1 | cut -d ' ' -f 2)
     __check_version java "$FABKIT_JAVA_VERSION_SUPPORTED" "$version"
 }
 
+__check_docker() {
+    if ! type -p docker &>/dev/null; then
+        logerr "docker is not installed. Version >= $FABKIT_DOCKER_VERSION_SUPPORTED is required"
+        exit 1
+    fi
+}
+
 __check_docker_version() {
+    __check_docker
+
     if docker info --format '{{json .}}' | grep -q "Cannot connect"; then
         logerr "Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?"
         exit 1

--- a/scripts/utils.sh
+++ b/scripts/utils.sh
@@ -23,7 +23,7 @@ __check_version() {
         if [[ "${installed_version[$i]//[aA-zZ]/}" -gt "${supported_version[$i]}" ]]; then
             break
         elif [[ "${installed_version[$i]//[aA-zZ]/}" -lt "${supported_version[$i]}" ]]; then
-            logerr "${cmd} >= $2 is required"
+            logerr "${cmd} >= $2 is required but installed version is: ${3//[aA-zZ]/}"
             exit 1
         fi
     done

